src/InvocationMocker.php

```diff
@@ -148,6 +158,54 @@ public function invoke(Invocation $invocation)
             return $returnValue;
         }
 
+        if ($this->returnValueGeneration === false) {
+            $exception = new ExpectationFailedException(
+                sprintf(
+                    "Return value inference disabled and no expectation set up for %s::%s()",
+                    $invocation->getClassName(),
+                    $invocation->getMethodName()
+                )
+            );
+
+            if (\strtolower($invocation->getMethodName()) === '__tostring') {
+                $this->addMatcher(
+                    new class($exception) extends StatelessInvocation
```

Comment:
Does this have to be an anonymous class?

Reply:
Absolutely not. My line of thinking was that I don't see any kind of possibly reuse for this workaround so I made it an inner class. I could extract a proper class and call it something like "DeferredError" or so.